tests/test_utils.py

```diff
@@ -107,16 +104,29 @@ def test_open_or_init_repository_initializes_repo_with_invalid_remote_branches(s
     assert open_or_init_repository(path, template) == repo
 
 
-def test_construct_keypair_defaults(Keypair: Mock):
+def test_construct_keypair_defaults(Keypair: Mock, MockPath: Mock, tmp_path: Path):
+    MockPath.return_value = tmp_path
+    public_key_path = tmp_path / 'id_rsa.pub'
+    public_key_path.touch()
+    private_key_path = tmp_path / 'id_rsa'
+    private_key_path.touch()
     construct_keypair()
-    user_home = os.path.expanduser('~')
-    Keypair.assert_called_once_with('git', f'{user_home}/.ssh/id_rsa.pub',
-                                    f'{user_home}/.ssh/id_rsa', '')
+    Keypair.assert_called_with('git', public_key_path,
+                               private_key_path, '')
 
 
-def test_construct_keypair(Keypair: Mock):
-    public_key_path = 'test-public_key_path'
-    private_key_path = 'test-private_key_path'
+@pytest.mark.parametrize('algorithm', ALGORITHMS.values())
+def test_construct_keypair(Keypair: Mock, tmp_path: Path, algorithm: dict[str, Path]):
+    public_key_path = tmp_path / algorithm['public']
+    public_key_path.touch()
+    private_key_path = tmp_path / algorithm['private']
+    private_key_path.touch()
     passphrase = 'test-passphrase'
-    construct_keypair(public_key_path, private_key_path, passphrase)
-    Keypair.assert_called_once_with('git', public_key_path, private_key_path, passphrase)
+    construct_keypair(tmp_path, passphrase)
+    Keypair.assert_called_with('git', public_key_path, private_key_path, passphrase)
+
+
+def test_construct_keypair_missing_key(MockPath: Mock, tmp_path: Path):
+    MockPath.return_value = tmp_path
+    with pytest.raises(KeypairMissingException):
+        construct_keypair()
```

Comment:
Added an additional test for when the error `KeypairMissingException` is raised due to missing files.